### ISSUE TITLE
Loading a key using AddSigningCredential - but still looking from config

### ISSUE THREAD

**Issue by Kirschstein** [IdentityServer/IdentityServer4#4000]
### Question
 I'm trying to get IdentityServer4 working on a Linux App Service on Azure. Linux doesn't have the same keystore as Windows, so the keys need to be fetched from disk. I've got the keys onto the file system, but IdentityServer seems still to be looking in config even though I've specified them in ConfigureApplication 

```csharp
    public void ConfigureServices(IServiceCollection services)
    {
        services.AddDefaultIdentity<IdentityUser>()
            .AddEntityFrameworkStores<ApplicationDbContext>();

        services.AddIdentityServer()
            .AddSigningCredential(GetCert())
            .AddApiAuthorization<IdentityUser, ApplicationDbContext>();

        services.AddAuthentication()
            .AddIdentityServerJwt();
    }

    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
    {
        app.UseAuthentication();
        app.UseIdentityServer(); // exception thrown here 
        app.UseAuthorization();
    }


    private X509Certificate2 GetCert()
    {
        // Linux App Service puts keys configured in the App Service in this directory
        // I have checked - the key is correctly placed here
        var bytes = System.IO.File.ReadAllBytes("/var/ssl/private/XYZTHUMBPRINTXYZ.p12");
        return new X509Certificate2(bytes);
    }
```

The exception being thrown is "invalid key type"

**Comment by leastprivilege** [comment]
The code looks OK to me. What kind of key is wrapped in the p12?

**Comment by Kirschstein** [comment]
@leastprivilege I've worked out what the issue is.

The call to ` .AddApiAuthorization<IdentityUser, ApplicationDbContext>();` is a helper method which under the surface calls a number of other methods, including a call to `AddSigningCredentials()` which then looks at app config for a location of a key. Whether or not the the call which specifies the cert is called before or after, IdentityServer will still look for a key specified from config.

To resolve it myself I have inlined the calls that `AddApiAuthorization` makes and deleted the call to `AddSigningCredentials()`, although it's a non-obvious reason as to why it's doing it from the surface.
